plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeConfig.java

```diff
@@ -489,4 +538,15 @@ public DeltaLakeConfig setQueryPartitionFilterRequired(boolean queryPartitionFil
         this.queryPartitionFilterRequired = queryPartitionFilterRequired;
         return this;
     }
+
+    @PostConstruct
+    public void validate()
```

Comment:
We disallow invalid config combinations in this project. Please take a look at other connectors, e.g. BigQueryConfig, IcebergConfig and etc.

Reply:
we typically use validation annotations for validation (eg `@AssertTrue`) rather than Guice's `@PostConstruct`.

Reply:
`delta.metastore.store-table-metadata-enabled=false` should be enough to disable the functionality
we provide defaults for two other parameters, and we should not burden the users with the need to disable them  (set to some non-default value accepted by validate)


ideally you'd have `delta.metastore.store-table-metadata-enabled` in one config class, and when set to true (or defaults to true), you'd bind additional config class with two other options. this is best from config validation perspective, but we often don't do that, as this would lead to much higher number of config classes (see `CachingHiveMetastoreConfig` for example).

in this case, i'd simply drop validation. the enabled true/false is pretty clear